google-cloud-bigquery/lib/google/cloud/bigquery/condition.rb

```diff
@@ -0,0 +1,218 @@
+# Copyright 2025 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+require "google/apis/bigquery_v2"
+
+module Google
+  module Cloud
+    module Bigquery
+      ##
+      # # Condition
+      #
+      # Represents a textual expression in the Common Expression Language (CEL) syntax.
+      # CEL is a C-like expression language. The syntax and semantics of CEL are documented
+      # at https://github.com/google/cel-spec
+      #
+      # Used to define condition for {Dataset::Access} rules
+      #
+      class Condition
+        ##
+        # Returns the textual representation of an expression in Common Expression Language syntax.
+        #
+        # @return [String] The expression of the condition.
+        #
+        # @example
+        #   condition = Google::Cloud::Bigquery::Condition.new(
+        #     "resource.name.startsWith('projects/my-project')"
+        #   )
+        #   puts condition.expression # => "resource.name.startsWith('projects/my-project')"
+        #
+        def expression
+          @expression
+        end
+
+        ##
+        # Sets the textual representation of an expression in Common Expression Language syntax.
+        #
+        # @param [String] expression The expression to set.
+        #
+        # @raise [ArgumentError] if the expression is nil or empty.
+        #
+        # @example
+        #   condition = Google::Cloud::Bigquery::Condition.new(
+        #     "resource.name.startsWith('projects/my-project')"
+        #   )
+        #   condition.expression = "document.summary.size() < 100"
+        #
+        def expression=expression
```

Comment:
Leave a space between the method name `expression=` and the method argument `expression`.

Further, it's probably a good idea for the argument name not to be the same name as the method. This is because the argument name then "shadows" the method name syntactically. (For example, it would be syntactically more difficult to recursively call the method if you ever needed to do that.) Typical practice for a setter method like this is either to use either a more generic name for the parameter (e.g. `def expression= val`) or use a prefix like `new_` (e.g. `def expression= new_expr`).